streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -317,7 +328,8 @@ public V apply(final K key, final Change<V> change) {
                                            final boolean leftOuter,
                                            final boolean rightOuter,
                                            final String joinMergeName,
-                                           final String internalQueryableName) {
+                                           final String internalQueryableName,
+                                           final KTableKTableJoinNode.KTableKTableJoinNodeBuilder kTableJoinNodeBuilder) {
```

Comment:
Ditto above, let's do the ` kTableJoinNodeBuilder.withJoinMergeProcessorParameters` in the caller, not here.

Reply:
As explained above, we need to pass the builder as a parameter to capture all information that is generated inside this method